crates/table/src/table.rs

```diff
@@ -426,7 +410,30 @@ impl Table {
             debug_assert!(deleted);
         }
 
-        self.delete_internal(blob_store, ptr)
+        // SAFETY: We've checked above that `self.is_row_present(ptr)`.
+        unsafe { self.delete_internal(blob_store, ptr) }
+    }
+
+    /// Deletes the row identified by `ptr` from the table.
```

Comment:
```suggestion
    /// Deletes the row identified by `ptr` from the table.
    ///
    /// The function `before` is run on the to-be-deleted row,
    /// if it is present, before deleting.
    /// This enables callers to extract the deleted row.
    /// E.g. applying deletes when squashing/merging a transaction into the committed state
    /// passes `|row| row.to_product_value()` as `before`
    /// so that the resulting `ProductValue`s can be passed to the subscription evaluator.
```